hydra_provider/src/data_feed_service/random/iterator.rs

```diff
@@ -0,0 +1,185 @@
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::Arc;
+use std::time::SystemTime;
+use rand::Rng;
+use rand::{rngs::StdRng, SeedableRng};
+
+use messages::{command::Command, health::Health, sensor::{Air, EkfNav1, EkfNav2, EkfQuat, GpsPos1, GpsPos2, GpsVel, Imu1, Imu2, Sensor, UtcTime}, state::{State, StateData}, Data, Log, Message};
+
+use crate::hydra_input::HydraInput;
+
+
+#[derive(Clone)]
+pub struct RandomDataFeedIterator {
+	pub is_running: Arc<AtomicBool>,
+}
+
+impl RandomDataFeedIterator {
+	pub async fn next(&mut self) -> Option<HydraInput> {
+		if !self.is_running.load(Ordering::Relaxed) {
+			return None
+		}
+
+		// SHOULD DO: move this outside. No need to initialize every iterator step
+		let mut std_rng = StdRng::from_entropy();
+
+		let time: fugit::Instant<u64, 1, 1000> = fugit::Instant::<u64, 1, 1000>::from_ticks(
+			SystemTime::now()
+				.duration_since(SystemTime::UNIX_EPOCH)
+				.unwrap()
+				.as_millis() as u64,
+		);
+	
+		// Convert timestamp to UTC time
+		let utc_time = UtcTime {
+			time_stamp: time.ticks() as u32,
+			day: (time.ticks() / 86400000) as i8,
+			year: (1970 + (time.ticks() / 86400000) / 365) as u16,
+			hour: ((time.ticks() % 86400000) / 3600000) as i8,
+			minute: ((time.ticks() % 3600000) / 60000) as i8,
+			month: ((time.ticks() % 31536000000) / 2592000000) as i8,
+			second: ((time.ticks() % 60000) / 1000) as i8,
+			nano_second: (time.ticks() % 1000) as i32 * 1000000,
+			gps_time_of_week: 0,
+			status: 0,
+		};
+	
+		let air = Air {
+			air_temperature: std_rng.gen(),
+			altitude: std_rng.gen(),
+			pressure_abs: std_rng.gen(),
+			pressure_diff: std_rng.gen(),
+			status: std_rng.gen(),
+			time_stamp: time.ticks() as u32,
+			true_airspeed: std_rng.gen(),
+		};
```

Comment:
Unrelated, but we just merged something in Hydra that could allow to automatically generate random sensor data here. I might open a PR for that.